### PR TITLE
Fix: couldn't delete file or image from user profile update

### PR DESCRIPTION
**Pull request by hasinur1997**
Couldn't delete file or image from user profile form.

Fix: https://bitbucket.org/wedevs/wpuf-pro/issues/219/image-and-file-upload-field-is-not

### CODE REVIEW

File: class/upload.php [hunk 1]
```diff
@@ -291,7 +291,7 @@ public function delete_file() {
         $attachment = get_post( $attachment_id );
 
         // post author or editor role
-        if ( get_current_user_id() == absint( $attachment->post_author ) || current_user_can( 'delete_private_pages' ) ) {
+        if ( get_current_user_id() == absint( $attachment->post_author ) || current_user_can( 'read' ) ) {
```
Comment (tareq1988) [inline class/upload.php:294]: @hasinur1997 the change you've made, doesn't it mean any logged in user can delete anyone's file? That's a serious security issue.

**Review by tareq1988 — CHANGES_REQUESTED** [review]

File: class/upload.php (hunk 1 above)
Comment (hasinur1997) [reply class/upload.php:294]: @tareq1988 Can you please suggest me a better way to solve this problem ?

Comment (tareq1988) [reply class/upload.php:294]: What’s the actual reason the deletion is failing?

Comment (hasinur1997) [reply class/upload.php:294]: If an image or file is given at the time of registration, it does not have an author id. When user update their profile and try to delete or update the image or file couldn't find the author id. Thats why it is failing to delete.

Comment (tareq1988) [reply class/upload.php:294]: Then this is not the solution.

When someone is registering, after registration when we have the user ID, we need to change the `post_author` to the newly registered user ID. That means, after registration, they will be the owner of those attachments. Then they will be able to delete their own attachments normally.

**Pull request closed without merging**